benchconnect/benchconnect/_augment.py

```diff
@@ -0,0 +1,97 @@
+from json import dumps, load, loads
+from pathlib import Path
+from typing import Callable
+
+import click
+from benchadapt.result import BenchmarkResult
+from benchadapt.run import BenchmarkRun
+
+
+def read_json_file(path: str) -> dict:
+    "Read a JSON field into a Python object"
+    with open(Path(path).resolve(), "r") as f:
+        return load(f)
+
+
+def augment_blob(json: dict, cls: Callable):
+    "Use dataclass to append default fields to a parsed JSON blob"
+    return cls(**json).to_publishable_dict()
+
+
+def print_json(json: dict) -> None:
+    "Print JSON nicely"
+    click.echo(dumps(json))
+
+
+def augment_and_print_blob(json: dict, cls: Callable) -> None:
+    "Append default fields to a blob and print as JSON to stdout"
+    augmented = augment_blob(json=json, cls=cls)
+    print_json(augmented)
+
+
+def augmentor(json: str, path: str, cls: Callable) -> None:
+    "For a given class, load data, augment it, and print it"
+    if json:
+        blob = loads(json)
+        augment_and_print_blob(json=blob, cls=cls)
+
+    elif path and Path(path).resolve().is_file():
+        blob = read_json_file(path=path)
+        augment_and_print_blob(json=blob, cls=cls)
+
+    elif path and Path(path).resolve().is_dir():
+        for file in Path(path).resolve().glob("*.json"):
+            blob = read_json_file(path=file)
+            augment_and_print_blob(json=blob, cls=cls)
```

Comment:
What will the workflow look like for "augment and then post a directory of JSONs"? Since `augment` returns ndjson to stdout, how can that be piped to `benchconnect post`? Would you mind putting an example of this somewhere?

Reply:
Hm, it would take a little munging. Maybe methods should take ndjson too?

Originally this was going to be an "all-in-one" situation where it augments and sends in one call, but that seemed opaque and hard to debug. But composability is nice

Reply:
This led down some weird rabbit holes around stdin and textio, but everything should now accept ndjson piped in (or just passed inline)